rmw_implementation/rmw_implementation-extras.cmake.in

```diff
@@ -15,69 +15,57 @@
-        "which was specified when the 'rmw_implementation' package was built.")
-    endif()
-  endif()
+  set(requested_rmw_implementation "${RMW_IMPLEMENTATION}")
+  set(requested_rmw_implementation_from "CMake")
+elseif(NOT "$ENV{RMW_IMPLEMENTATION}" STREQUAL "")
+  set(requested_rmw_implementation "$ENV{RMW_IMPLEMENTATION}")
+  set(requested_rmw_implementation_from
+    "environment variable 'RMW_IMPLEMENTATION'")
+else()
+  set(requested_rmw_implementation "@RMW_IMPLEMENTATION@")
+  set(requested_rmw_implementation_from
+    "the default when @PROJECT_NAME@ was built")
 endif()
 
-if(NOT "$ENV{RMW_IMPLEMENTATION}" STREQUAL "")
-  set(requested_rmw_implementation $ENV{RMW_IMPLEMENTATION})
-  if(@RMW_IMPLEMENTATION_SUPPORTS_POCO@)
-    if(NOT requested_rmw_implementation IN_LIST available_rmw_implementations)
-      message(FATAL_ERROR
-        "The RMW implementation has been specified as "
-        "'${requested_rmw_implementation}' through the environment variable "
-        "'RMW_IMPLEMENTATION', "
-        "however it is not in the list of supported rmw implementations, "
-        "which was specified when the 'rmw_implementation' package was built.")
-    endif()
-  else()
-    if(NOT requested_rmw_implementation STREQUAL default_rmw_implementation)
-      message(FATAL_ERROR
-        "The RMW implementation has been specified as "
-        "'${requested_rmw_implementation}' "
-        "through the environment variable 'RMW_IMPLEMENTATION', "
-        "however this needs to match the RMW implementation "
-        "'${default_rmw_implementation}', "
-        "which was specified when the 'rmw_implementation' package was built.")
-    endif()
+if(@RMW_IMPLEMENTATION_DISABLE_RUNTIME_SELECTION@)
+  message(STATUS "Using RMW implementation '@RMW_IMPLEMENTATION@'")
+  if(NOT requested_rmw_implementation STREQUAL "@RMW_IMPLEMENTATION@")
+    message(FATAL_ERROR
+      "The RMW implementation has been specified as "
+      "'${requested_rmw_implementation}' via "
+      "${requested_rmw_implementation_from}, but rmw_implementation was built "
+      "only with support for '@RMW_IMPLEMENTATION@'."
+    )
+  endif()
+  find_package("@RMW_IMPLEMENTATION@" REQUIRED)
+
+  # TODO should never need definitions and include dirs?
+  list(APPEND rmw_implementation_DEFINITIONS
+    ${@RMW_IMPLEMENTATION@_DEFINITIONS})
+  list(APPEND rmw_implementation_INCLUDE_DIRS
+    ${@RMW_IMPLEMENTATION@_INCLUDE_DIRS})
+  list(APPEND rmw_implementation_LIBRARIES
+    ${@RMW_IMPLEMENTATION@_LIBRARIES})
+else()
+
+  get_available_rmw_implementations(available_rmw_implementations)
+
+  if(NOT requested_rmw_implementation IN_LIST available_rmw_implementations)
+    message(FATAL_ERROR
+      "The RMW implementation has been specified as "
+      "'${requested_rmw_implementation}' via "
+      "${requested_rmw_implementation_from}, but it is not available at this "
+      "time.\n Make sure it does not expect a typesupport or other build "
```

Comment:
> Make sure it does not expect a typesupport or other build dependencies.

I don't understand the meaning / purpose of this part of the message. The specified RMW impl isn't available - what does the type support and/or other dependencies have to do with that?

Reply:
I wanted to make sure that things like rmw_fastrtps_dynamic_cpp and rmw_cyclonedds_cpp were not rejected by the build system if they were unavailable at the time this was built. I think you’re right that, now that this is a fatal error, that language is unnecessary